site/src/job_queue/mod.rs

```diff
@@ -174,12 +178,63 @@ pub async fn build_queue(
     Ok(queue)
 }
 
+/// From a benchmark_request create all the required jobs
+pub fn create_benchmark_jobs(
+    benchmark_request: &BenchmarkRequest,
+) -> anyhow::Result<Vec<BenchmarkJob>> {
+    anyhow::ensure!(
+        benchmark_request.tag().is_some(),
+        "Benchmark request has no tag"
+    );
+
+    let backends = benchmark_request.backends()?;
+    let profiles = benchmark_request.profiles()?;
+    let mut jobs = vec![];
+
+    // Target x benchmark_set x backend x profile -> BenchmarkJob
+    for target in Target::all() {
+        for benchmark_set in 0..benchmark_set_count(
+            collector::compile::benchmark::target::Target::from_db_target(&target),
+        ) {
+            for backend in backends.iter() {
+                for profile in profiles.iter() {
+                    let job = BenchmarkJob::new(
+                        target,
+                        *backend,
+                        *profile,
+                        benchmark_request.tag().unwrap(),
+                        benchmark_set as u32,
+                        Utc::now(),
+                        BenchmarkJobStatus::Queued,
+                    );
+                    jobs.push(job);
```

Comment:
Here we can also create the jobs for the parent. It will have the same parameters, just with the parent tag.